### PR TITLE
feat: TAP-1315 You can configure a proxy server when configuring the …

### PR DESCRIPTION
**Pull request by mnianqi**
…SMTP service

### CODE REVIEW

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_reliability_rating**, ActualValue: **3**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-1315-proxy-SMTP-service&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_reliability_rating**, ActualValue: **3**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-1315-proxy-SMTP-service&id=tapdata

File: iengine/iengine-app/src/main/java/io/tapdata/common/BaseEmailExecutor.java [hunk 1]
```diff
@@ -52,13 +54,28 @@ protected Mailer initMailer() {
...
-						.withProperty("mail.smtp.ssl.enable", StringUtils.equalsAnyIgnoreCase(encryptTypeSett.getValue(), "SSL") ? "true" : "false")
-						.withProperty("mail.smtp.starttls.enable", StringUtils.equalsAnyIgnoreCase(encryptTypeSett.getValue(), "TLS") ? "true" : "false")
-						.buildMailer();
+				String proxyHost = null;
+				String proxyPort = null;
+				if (null != proxyHostSett && null != proxyPortSett){
+					proxyHost = proxyHostSett.getValue();
+					proxyPort = proxyPortSett.getValue();
+				}
+				if (StringUtils.isNotBlank(proxyHost) && StringUtils.isNotBlank(proxyPort)) {
```
Comment (ply0011) [inline iengine/iengine-app/src/main/java/io/tapdata/common/BaseEmailExecutor.java:63]: 尝试减少重复代码

File: manager/tm/src/main/java/com/tapdata/tm/Settings/service/SettingsServiceImpl.java [hunk 2]
```diff
@@ -317,9 +323,17 @@ public void testSendMail(TestMailDto testMailDto) {
 
     }
 
-    private MailAccountDto getMailAccount(TestMailDto testMailDto) {
+    protected MailAccountDto getMailAccount(TestMailDto testMailDto) {
 
         String[] split = testMailDto.getEmail_Receivers().split(",");
+        String proxyHost = testMailDto.getSMTP_Proxy_Host();
+        String proxyPort = testMailDto.getSMTP_Proxy_Port();
+        if (StringUtils.isNotBlank(proxyHost) && StringUtils.isNotBlank(proxyPort)) {
```
Comment (ply0011) [inline manager/tm/src/main/java/com/tapdata/tm/Settings/service/SettingsServiceImpl.java:331]: 尝试减少判断分支

File: manager/tm/src/main/java/com/tapdata/tm/Settings/service/SettingsServiceImpl.java [hunk 3]
```diff
@@ -131,6 +133,10 @@ public MailAccountDto getMailAccount(String userId) {
             }
         }
 
+        if (StringUtils.isNotBlank(proxyHost) && StringUtils.isNotBlank(proxyPort)) {
```
Comment (ply0011) [inline manager/tm/src/main/java/com/tapdata/tm/Settings/service/SettingsServiceImpl.java:136]: 尝试减少判断分支

**Review by ply0011 — CHANGES_REQUESTED** [review]

**Review by ply0011 — APPROVED** [review]

**Pull request merged**